ksm.go

```diff
@@ -32,14 +33,54 @@ type ksmSetting struct {
 	scanIntervalMs     uint32
 }
 
+type ksmSettingKnob string
+
 const (
-	ksmSlug       = "slug"
-	ksmSlow       = "slow"
-	ksmStandard   = "standard"
-	ksmAggressive = "aggressive"
+	ksmSlug       ksmSettingKnob = "slug"
+	ksmSlow                      = "slow"
+	ksmStandard                  = "standard"
+	ksmAggressive                = "aggressive"
 )
 
-var ksmSettings = map[string]ksmSetting{
+func (k ksmSettingKnob) String() string {
+	switch k {
+	case ksmSlug:
+		return "slug"
+	case ksmSlow:
+		return "slow"
+	case ksmStandard:
+		return "standard"
+	case ksmAggressive:
+		return "aggressive"
+	}
+
+	return ""
+}
+
+func (k *ksmSettingKnob) Set(value string) error {
```

Comment:
just as we mentioned on irc, we should probably have some docs somewhere that note the proxy has some dials you can turn. We probably need an 'experts guide to tweaking CC' in the runtime docs place really.